bundle/tests/variables/env_overrides/databricks.yml

```diff
@@ -6,6 +6,10 @@ variables:
   b:
     description: required variable
 
+  d:
+    description: variable with lookup
+    lookup: "clusters: some-cluster"
```

Comment:
Using this for a bit, I realized the prefixed string is odd given that we have a structured for everything else. It also doesn't help discovery/documentation, as we can't autocomplete this.

I suspect this is a remnant of earlier iterations where we lumped this into the `default` key with a lookup string. Then we later figured to break it out into a separate key but didn't revisit whether it should be a string or not.

However, if we break out the lookup type into an object with a key for every resolver, we get discovery and documentation for free.

Reply:
Added structured lookup field